src/lib/lockfile.ts

```diff
@@ -0,0 +1,34 @@
+import * as fs from 'fs';
+import { dirname } from 'path';
+
+import { exec, unlinkAll } from './fs-utils';
+
+// Equivalent to `drwxrwxrwt`
+const STICKY_WRITE_PERMISSIONS = parseInt('1777', 8);
```

Comment:
You can do octal literals in javascript by prefixing with a `0`. In this case 

```
const STICKY_WRITE_PERMISSIONS = 01777
```

Reply:
Changed to 0o1777 per `Octal literals are not available when targeting ECMAScript 5 and higher. Use the syntax '0o1777'.ts(1085)`